### PR TITLE
moved deadline commands to be made in a assignment channel

### LINKED ISSUE #1: Review Qs/Ans
Improve the review questions to be personalized per student. Possible categories of questions.

- [ ]     Allow students to have their own set of review questions sent to their DM
- [ ]     Set it up so that the student can mark questions as not needing to be reviewed anymore.
- [ ]     Mark questions as needing to be reviewed more.
- [ ]     Post questions with time interval setting
- [ ]     Add categories to questions so students can get questions for a specific category
- [ ]     Allow students to answer with poll before posting answers
- [ ]     Allow instructors to add multiple review questions and answers at once
- [ ]     Allow all review questions to be printed at once

https://github.com/SE21-Team2/ClassMateBot/issues/21

### LINKED ISSUE #2: Allow students to have their own set of review questions sent to their DM

### PR DESCRIPTION
**Pull request by jhnguye4**
### Checklist

- [ ] Did you test your changes locally?  
- [ ] Did your changes pass all existing tests?  
- [ ] Do test cases exist for your changes? If not, please make an issue.

Note: ensure that all new code has test coverage! 

----

#### Closing Issues

Use `closes #XXXX` if your PR resolves an issue.
To close multiple issues, use: `resolves #1, resolves #2, resolves...`

#### Contributors

Use `@author`

### CODE REVIEW

**Review by snapcat — CHANGES_REQUESTED** [review]
Please create a new branch on this repo and copy and paste your changes there. The reason the tests are failing is likely because of the differences between the workflow .env

File: cogs/deadline.py [hunk 1]
```diff
@@ -34,6 +34,10 @@ def __init__(self, bot):
                       help="put in current time to get offset needed for proper "
                            "datetime notifications $timenow MMM DD YYYY HH:MM ex. $timenow SEP 25 2024 17:02")
     async def timenow(self, ctx, *, date: str):
+        if not ctx.channel.name == 'assignments':
+            await ctx.author.send('Please send questions to the #assignments channel.')
```
Comment (snapcat) [inline cogs/deadline.py:38]: 'Please send questions to the #assignments channel.'

Should be:

'Please use this command inside the #assignments channel.'

File: cogs/deadline.py [hunk 2]
```diff
@@ -84,6 +88,11 @@ async def timenow_error(self, ctx, error):
                       help="add homework and due-date $addhw CLASSNAME HW_NAME MMM DD YYYY optional(HH:MM) "
                       "ex. $addhw CSC510 HW2 SEP 25 2024 17:02")
     async def duedate(self, ctx, coursename: str, hwcount: str, *, date: str):
+        if not ctx.channel.name == 'assignments':
+            await ctx.author.send('Please send questions to the #assignments channel.')
```
Comment (snapcat) [inline cogs/deadline.py:92]: 'Please send questions to the #assignments channel.'

Should be:

'Please use this command inside the #assignments channel.'

File: cogs/deadline.py [hunk 3]
```diff
@@ -147,6 +156,10 @@ async def duedate_error(self, ctx, error):
                       help="delete a specific reminder using course name and homework name using "
                       "$deletereminder CLASSNAME HW_NAME ex. $deletereminder CSC510 HW2 ")
     async def deleteReminder(self, ctx, courseName: str, hwName: str):
+        if not ctx.channel.name == 'assignments':
+            await ctx.author.send('Please send questions to the #assignments channel.')
```
Comment (snapcat) [inline cogs/deadline.py:160]: 'Please send questions to the #assignments channel.'

Should be:

'Please use this command inside the #assignments channel.'

File: cogs/deadline.py [hunk 4]
```diff
@@ -197,6 +210,10 @@ async def deleteReminder_error(self, ctx, error):
                       help="update the assignment date. $changeduedate CLASSNAME HW_NAME MMM DD YYYY optional(HH:MM) "
                       "ex. $changeduedate CSC510 HW2 SEP 25 2024 17:02 ")
     async def changeduedate(self, ctx, classid: str, hwid: str, *, date: str):
+        if not ctx.channel.name == 'assignments':
+            await ctx.author.send('Please send questions to the #assignments channel.')
```
Comment (snapcat) [inline cogs/deadline.py:214]: 'Please send questions to the #assignments channel.'

Should be:

'Please use this command inside the #assignments channel.'

File: cogs/deadline.py [hunk 5]
```diff
@@ -251,6 +268,10 @@ async def changeduedate_error(self, ctx, error):
     @commands.command(name="duethisweek", pass_context=True,
                       help="check all the homeworks that are due this week $duethisweek")
     async def duethisweek(self, ctx):
+        if not ctx.channel.name == 'assignments':
+            await ctx.author.send('Please send questions to the #assignments channel.')
```
Comment (snapcat) [inline cogs/deadline.py:272]: 'Please send questions to the #assignments channel.'

Should be:

'Please use this command inside the #assignments channel.'

File: cogs/deadline.py [hunk 6]
```diff
@@ -293,6 +314,10 @@ async def duethisweek_error(self, ctx, error):
     # -----------------------------------------------------------------------------------------------------------------
     @commands.command(name="duetoday", pass_context=True, help="check all the homeworks that are due today $duetoday")
     async def duetoday(self, ctx):
+        if not ctx.channel.name == 'assignments':
+            await ctx.author.send('Please send questions to the #assignments channel.')
```
Comment (snapcat) [inline cogs/deadline.py:318]: 'Please send questions to the #assignments channel.'

Should be:

'Please use this command inside the #assignments channel.'

File: cogs/deadline.py [hunk 7]
```diff
@@ -332,6 +357,10 @@ async def duetoday_error(self, ctx, error):
                       help="check all the homeworks that are due for a specific course $coursedue coursename "
                       "ex. $coursedue CSC505")
     async def coursedue(self, ctx, courseid: str):
+        if not ctx.channel.name == 'assignments':
+            await ctx.author.send('Please send questions to the #assignments channel.')
```
Comment (snapcat) [inline cogs/deadline.py:361]: 'Please send questions to the #assignments channel.'

Should be:

'Please use this command inside the #assignments channel.'

File: cogs/deadline.py [hunk 8]
```diff
@@ -371,6 +400,10 @@ async def coursedue_error(self, ctx, error):
     # ---------------------------------------------------------------------------------
     @commands.command(name="listreminders", pass_context=True, help="lists all reminders")
     async def listreminders(self, ctx):
+        if not ctx.channel.name == 'assignments':
+            await ctx.author.send('Please send questions to the #assignments channel.')
```
Comment (snapcat) [inline cogs/deadline.py:404]: 'Please send questions to the #assignments channel.'

Should be:

'Please use this command inside the #assignments channel.'

File: cogs/deadline.py [hunk 9]
```diff
@@ -409,6 +442,10 @@ async def listreminders_error(self, ctx, error):
 
     @commands.command(name="clearreminders", pass_context=True, help="deletes all reminders")
     async def clearallreminders(self, ctx):
+        if not ctx.channel.name == 'assignments':
+            await ctx.author.send('Please send questions to the #assignments channel.')
```
Comment (snapcat) [inline cogs/deadline.py:446]: 'Please send questions to the #assignments channel.'

Should be:

'Please use this command inside the #assignments channel.'

**Review by snapcat — CHANGES_REQUESTED** [review]
'Please send questions to the #assignments channel.'

Should be:

'Please use this command inside the #assignments channel.'

**Comment by snapcat** [comment]
Converted to draft!

File: cogs/deadline.py [hunk 10]
```diff
@@ -251,6 +268,10 @@ async def changeduedate_error(self, ctx, error):
     @commands.command(name="duethisweek", pass_context=True,
                       help="check all the homeworks that are due this week $duethisweek")
     async def duethisweek(self, ctx):
+        if not ctx.channel.name == 'assignments':
+            await ctx.author.send('Please send questions to the #assignments channel.')
+            await ctx.message.delete()
+            return
```
Comment (snapcat) [inline cogs/deadline.py:274]: duethisweek should be useable from anywhere.

```suggestion
```

File: cogs/deadline.py [hunk 11]
```diff
@@ -34,6 +34,10 @@ def __init__(self, bot):
                       help="put in current time to get offset needed for proper "
                            "datetime notifications $timenow MMM DD YYYY HH:MM ex. $timenow SEP 25 2024 17:02")
     async def timenow(self, ctx, *, date: str):
+        if not ctx.channel.name == 'assignments':
+            await ctx.author.send('Please send questions to the #assignments channel.')
+            await ctx.message.delete()
+            return
```
Comment (snapcat) [inline cogs/deadline.py:40]: timenow should be used anywhere.

```suggestion
```

File: cogs/deadline.py (hunk 3 above)
Comment (snapcat) [inline cogs/deadline.py:160]: Should be:
```suggestion
            await ctx.author.send('Please use this command inside the #assignments channel.')
```

File: cogs/deadline.py (hunk 4 above)
Comment (snapcat) [inline cogs/deadline.py:214]: Should be:
```suggestion
            await ctx.author.send('Please use this command inside the #assignments channel.')
```

File: cogs/deadline.py [hunk 12]
```diff
@@ -293,6 +314,10 @@ async def duethisweek_error(self, ctx, error):
     # -----------------------------------------------------------------------------------------------------------------
     @commands.command(name="duetoday", pass_context=True, help="check all the homeworks that are due today $duetoday")
     async def duetoday(self, ctx):
+        if not ctx.channel.name == 'assignments':
+            await ctx.author.send('Please send questions to the #assignments channel.')
+            await ctx.message.delete()
+            return
```
Comment (snapcat) [inline cogs/deadline.py:320]: duetoday should be usable from anywhere.

```suggestion

```

File: cogs/deadline.py [hunk 13]
```diff
@@ -332,6 +357,10 @@ async def duetoday_error(self, ctx, error):
                       help="check all the homeworks that are due for a specific course $coursedue coursename "
                       "ex. $coursedue CSC505")
     async def coursedue(self, ctx, courseid: str):
+        if not ctx.channel.name == 'assignments':
+            await ctx.author.send('Please send questions to the #assignments channel.')
+            await ctx.message.delete()
+            return
```
Comment (snapcat) [inline cogs/deadline.py:363]: coursedue should be usable from anywhere.

```suggestion

```

File: cogs/deadline.py [hunk 14]
```diff
@@ -371,6 +400,10 @@ async def coursedue_error(self, ctx, error):
     # ---------------------------------------------------------------------------------
     @commands.command(name="listreminders", pass_context=True, help="lists all reminders")
     async def listreminders(self, ctx):
+        if not ctx.channel.name == 'assignments':
+            await ctx.author.send('Please send questions to the #assignments channel.')
+            await ctx.message.delete()
+            return
```
Comment (snapcat) [inline cogs/deadline.py:406]: listreminders should be usable from anywhere

```suggestion

```

**Review by snapcat — CHANGES_REQUESTED** [review]
Also check with @etracey7 since she's also working on deadline.py!

**Comment by peeyush10234** [comment]
The task is already being addressed in another PR.

**Pull request closed without merging**